airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/java/io/airbyte/integrations/base/destination/typing_deduping/BaseTypingDedupingTest.java

```diff
@@ -825,6 +825,25 @@ protected AirbyteDestination startSync(final ConfiguredAirbyteCatalog catalog,
 
     destination.start(destinationConfig, jobRoot, Collections.emptyMap());
 
+    // In the background, read messages from the destination until it terminates. We need to clear
+    // stdout in real time, to prevent the buffer from filling up and blocking the destination.
+    // TODO Eventually we'll want to somehow extract the state messages while a sync is running, to
+    // verify checkpointing.
+    final ExecutorService messageHandler = Executors.newSingleThreadExecutor(
+        // run as a daemon thread just in case we run into an exception or something
+        r -> {
+          final Thread t = Executors.defaultThreadFactory().newThread(r);
+          t.setDaemon(true);
+          return t;
+        }
+    );
+    messageHandler.submit(() -> {
+      while (!destination.isFinished()) {
+        destination.attemptRead();
```

Comment:
I didn't see another call for attemptRead so seems safe but just a note that attemptRead isn't thread safe.

Reply:
yep it's only called from here. I'll add a comment though